### ISSUE TITLE
SVG export in dark mode fix does not work on iOS

### ISSUE THREAD

**Issue by zsviczian** [excalidraw/excalidraw#4301]
#4285 did not fix issue on iOS.

**Comment by zsviczian** [comment]
Ok. This is a different issue. Looking for the root cause. If I upload this SVG that has the inversion correctly, then the colors are ok.
[F1.zip](https://github.com/excalidraw/excalidraw/files/7576859/F1.zip)

**Comment by dwelle** [comment]
Unclear what's the issue here. SVG exports or canvas? In the GIF above, is that a PNG/JPG or a SVG? Note that SVG images _are_ inverted on canvas right now, unlike PNG/JPGs. The reason being is that most commonly, SVGs are B&W, so reverting is necessary for them to be visible on the dark background. A better heuristic may be implemented later.

**Comment by zsviczian** [comment]
In the screen capture that is a PNG image. The zip is an example of an SVG. The issue is behavior with the same two images is different on iOS than on LINUX, Windows, Android. Both these images show correctly on those three and are inverted on iOS.

**Comment by dwelle** [comment]
Of I forgot, that canvas filter isn't supported on Safari: https://caniuse.com/?search=CanvasRenderingContext2D.filter, so we'll need to do it differently.

**Comment by zsviczian** [comment]
ok. So for Apple users, if they want to use images, they can only use the light mode. Seems only IE and Safari do not support context filtering.

**Comment by zsviczian** [comment]
this or a package similar might be an interesting option:
https://github.com/davidenke/context-filter-polyfill

**Comment by dwelle** [comment]
Yes, I've seen that one, but never used yet. The way I understand it, it works asynchronously, thus not really suited for our type of render loop — though we could render the images on the side and cache the output (the caching we do anyway).

Another way would be to directly manipulate the bitmap data and invert, hue-rotate, and saturate the pixels ourselves (then render it and cache it).

**Comment by dwelle** [comment]
fixed by https://github.com/excalidraw/excalidraw/pull/10578
